### PR TITLE
multipart/form-data support added for file upload operations

### PR DESCRIPTION
**Pull request by sergiishamrai-okta**
Need to have the ability to upload files, so POST multipart/form-data support added

Origin: [OKTA-448365](https://oktainc.atlassian.net/browse/OKTA-448365)

Related PR: https://github.com/okta/okta-sdk-java/pull/679

### CODE REVIEW

File: http/okhttp/src/main/java/com/okta/commons/http/okhttp/OkHttpRequestExecutor.java [hunk 1]
```diff
@@ -126,7 +135,21 @@ public Response executeRequest(Request request) throws HttpException {
                 okRequestBuilder.head();
                 break;
             case POST:
-                okRequestBuilder.post(new InputStreamRequestBody(request.getBody(), request.getHeaders().getContentType()));
+                if(isMultipartFormDataForFileUploading) {
+                    String fileLocation = fetchHeaderValueAndRemoveIfExists(request, "x-fileLocation");
```
Comment (arvindkrishnakumar-okta) [inline http/okhttp/src/main/java/com/okta/commons/http/okhttp/OkHttpRequestExecutor.java:139]: Just for my understanding: why are we removing the headers `x-fileFormDataName` & `x-fileLocation` here?

File: http/okhttp/src/main/java/com/okta/commons/http/okhttp/OkHttpRequestExecutor.java [hunk 2]
```diff
@@ -147,6 +170,15 @@ public Response executeRequest(Request request) throws HttpException {
         }
     }
 
+    private String fetchHeaderValueAndRemoveIfExists(Request request, String headerName) {
```
Comment (arvindkrishnakumar-okta) [inline http/okhttp/src/main/java/com/okta/commons/http/okhttp/OkHttpRequestExecutor.java:173]: Nit: Does `removeHeaderIfPresent(..)` sound like a better method name?

File: http/okhttp/src/test/groovy/com/okta/commons/http/okhttp/OkHttpRequestExecutorTest.groovy [hunk 3]
```diff
@@ -123,6 +123,89 @@ class OkHttpRequestExecutorTest {
...
+        assertThat response.httpStatus, is(200)
+        assertThat response.isClientError(), is(false)
+        assertThat response.isError(), is(false)
+        assertThat response.isServerError(), is(false)
+
+        verify(requestAuthenticator).authenticate(request)
+    }
+
+    @Test
+    void testExecuteRequestPostForFileUploading() {
```
Comment (arvindkrishnakumar-okta) [inline http/okhttp/src/test/groovy/com/okta/commons/http/okhttp/OkHttpRequestExecutorTest.groovy:167]:
```suggestion
    void testExecuteRequestPostForMultipartFileUploading() {
```

File: http/okhttp/src/main/java/com/okta/commons/http/okhttp/OkHttpRequestExecutor.java (hunk 1 above)
Comment (sergiishamrai-okta) [reply http/okhttp/src/main/java/com/okta/commons/http/okhttp/OkHttpRequestExecutor.java:139]: These headers are not required to perform file uploading. They are used to pass info from SDK to the HTTP client.

File: http/okhttp/src/main/java/com/okta/commons/http/okhttp/OkHttpRequestExecutor.java (hunk 2 above)
Comment (sergiishamrai-okta) [reply http/okhttp/src/main/java/com/okta/commons/http/okhttp/OkHttpRequestExecutor.java:173]: the method receives value from headers if the key exists and removes it. IMHO removeHeaderIfPresent does not describe fully what method does.

**Review by arvindkrishnakumar-okta — APPROVED** [review]

**Comment by bdemers** [comment]
Did anything need to change for Apache HttpClient?

**Comment by sergiishamrai-okta** [comment]
> Did anything need to change for Apache HttpClient?

We use OkHttp3 client in Java Mgmt SDK so no changes are required in Apache HttpClient

**Comment by bdemers** [comment]
@sergiishamrai-okta customers have a choice between ok-http and Apache http-client.  Any features added to one client must be added to the other.

**Review by arvindkrishnakumar-okta — APPROVED** [review]
LGTM

**Pull request merged**